ssa.go

```diff
@@ -186,9 +187,13 @@ func ReadFromSSA(i io.Reader) (o *Subtitles, err error) {
 
 		// Split on ":"
 		var split = strings.Split(line, ":")
-		if len(split) < 2 {
-			err = fmt.Errorf("astisub: line '%s' should contain at least one ':'", line)
-			return
+		if len(split) < 2 || split[0] == "" {
+			switch sectionName {
+			case ssaSectionNameScriptInfo, ssaSectionNameStyles: // Do nothing
```

Comment:
Can you provide an example for this case?

Reply:
Here's the fragments I had problems with:
```
[V4+ Styles]
Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, Alignment, MarginL, MarginR, MarginV, Encoding
Style: Default,MS UI Gothic,90,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100,100,5,0,1,2,2,1,10,10,10,0
//
Style: Rubi,MS UI Gothic,50,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,1,2,2,1,10,10,10,0
//

[Events]
```
```
Dialogue: 0,0:23:40.69,0:23:45.56,Default,,0000,0000,0000,,東福寺駅近くにある会社からの人生の訓辞なのかなぁ～
;任天堂京都本社

:次回標題
Dialogue: 0,0:23:45.79,0:23:47.20,Default,,0000,0000,0000,,次回　「願望」
```
The specification says: `SSA will discard any lines it doesn't understand`.
and also I wanted to make the behaviour closer to libass.
Libass only reports about ill-formed lines if they are in [Fonts] or [Events] sections. Otherwise, it discard them like the rest.

Reply:
OK makes sense.

Could you add at least one line not understood by the SSA parser in the `testdata/example-in.ssa` so that this case is covered by the tests?

Reply:
Hmm... this event line
`Unknown descriptor: discarded`
fails with this error:
`astisub: building new ssa event failed: astisub: content has 1 items whereas style format has 10 items`
So maybe instead of the `|| split[0] == ""` check, we should check if split[0] is an unknown descriptor.

Reply:
I think this check is fine to avoid lines without a `:`.

But as you mention there should be an additionnal check for the descriptor. I would place it [here](https://github.com/asticode/go-astisub/blob/master/ssa.go#L210) and would check the value of `header` based on the value of `sectionName` and discard lines that don't match the correct `header` value.

Does that make sense?

Reply:
The specific `|| split[0] == ""` check is not to avoid lines without a `:`, it is to avoid lines with an empty descriptor (the `:次回標題` line above). Anyway, handling of unknown descriptors can be added later after the merge of this PR. go-astisub now works with subtitles I needed, and that's enough for me for now.